### DISCUSSION TITLE
片方のキーボードが認識されない

### DISCUSSION THREAD

**Discussion by Ryahoobb** [Yowkees/keyball#301] [Q&A] [answered]
こちらと同様の状況となっており、質問させてください。

 [TRSケーブルで繋いだ方のキーボードが認識されない](https://github.com/Yowkees/keyball/discussions/245) https://github.com/Yowkees/keyball/discussions/245

QAに記載のようにTRRSソケットや、ソケット上のダイオード、ジャンパのはんだ付けをやり直したのですがうまくいかず、確認すべき点をご教示いただけますと幸いです。

【状況】
・左手トラックボール仕様
・本番用のファームウェアを書き込み、全てのキースイッチ、トラックボールが動作することを確認済
・LEDは未着用
・USBで直接繋いでいれば確認できますが、片方をUSBで繋ぎ、TRRSケーブルで繋いでもTRRSでケーブルで繋いだ方のキー、トラックボールが反応しません。（たとえば右を繋ぐと左のキーとボールが反応しない、左を繋ぐと右のキーが反応しない）。
・Pro Micro Type-C版 を使用 https://shop.yushakobo.jp/products/3905?_pos=8&_sid=9e9bd7147&_ss=r

![PXL_20230811_135725527](https://github.com/Yowkees/keyball/assets/37834735/471cc9dd-44cf-43d1-9ee0-6dca8d16343b)

右手の裏側です。
![PXL_20230811_140957422](https://github.com/Yowkees/keyball/assets/37834735/d70fa8ed-3737-482f-ad5f-56c34a49bbeb)

左手の裏側です。
![PXL_20230811_141014958](https://github.com/Yowkees/keyball/assets/37834735/b242ce25-3ad1-47f5-b853-77b8daec4b84)

**Comment by Yowkees** [comment, marked as answer]
ご連絡ありがとうございます。

症状的に左右のDATA線が導通できていないようです。

下記添付画像の緑矢印で示した半田付けですが、丸々としていてテンプラしているように見えます。
こちらを再度半田付けしていただき、半田が基板のパッドに馴染んで、山型になるように修正してみてください。

また、オレンジ点で示した位置も半田付けをご確認ください。
左手側も同じ位置にDATA端子があります。左右のオレンジ色点どうしが導通していれば正常に通信すると思います。

お手数ですがご対応お願いいたします。
![KB44_datafix](https://github.com/Yowkees/keyball/assets/84209772/96727231-8eb7-47fa-bdba-15a796f0bcf6)

**Reply by Ryahoobb** [reply]
さっそくご指摘ありがとうございます。
ご指摘箇所のはんだ付けを見直し、無事左右ともに認識するようになりました。

また、途中でトラックボールが認識されない状態になったのすが、過去ログを参考にL字コンスルーやセンサーICのはんだ付けを直すことで無事すべて動作するようになりました。
大切につかわせていただきます。丁寧に解説いただきありがとうございました。
